Failing generation on beta validation pipeline due to missing IDataSourceWithReferenceRequestBuilder

From the pipeline link below
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=73749&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=006e6f7c-8314-5e02-a3df-483f3a1f4cfa

Generation seems to fail in dotnet due to the compile error

```
##[error]src/Microsoft.Graph/Generated/securitynamespace/requests/EdiscoverySearchCustodianSourcesCollectionWithReferencesRequestBuilder.cs(56,16): Error CS0246: The type or namespace name 'IDataSourceWithReferenceRequestBuilder' could not be found (are you missing a using directive or an assembly reference?)
/home/vsts/work/1/s/msgraph-sdk-dotnet/src/Microsoft.Graph/Generated/securitynamespace/requests/EdiscoverySearchCustodianSourcesCollectionWithReferencesRequestBuilder.cs(56,16): error CS0246: The type or namespace name 'IDataSourceWithReferenceRequestBuilder' could not be found (are you missing a using directive or an assembly reference?) [/home/vsts/work/1/s/msgraph-sdk-dotnet/src/Microsoft.Graph/Microsoft.Graph.csproj]
##[error]src/Microsoft.Graph/Generated/securitynamespace/requests/IEdiscoverySearchCustodianSourcesCollectionWithReferencesRequestBuilder.cs(37,9): Error CS0246: The type or namespace name 'IDataSourceWithReferenceRequestBuilder' could not be found (are you missing a using directive or an assembly reference?)
/home/vsts/work/1/s/msgraph-sdk-dotnet/src/Microsoft.Graph/Generated/securitynamespace/requests/IEdiscoverySearchCustodianSourcesCollectionWithReferencesRequestBuilder.cs(37,9): error CS0246: The type or namespace name 'IDataSourceWithReferenceRequestBuilder' could not be found (are you missing a using directive or an assembly reference?) [/home/vsts/work/1/s/msgraph-sdk-dotnet/src/Microsoft.Graph/Microsoft.Graph.csproj]
```

Ideally the request builder should exist in the target namespace and will need to be investigated to understand the issue.

--- a/src/GraphODataTemplateWriter/Extensions/OdcmModelExtensions.cs
+++ b/src/GraphODataTemplateWriter/Extensions/OdcmModelExtensions.cs
@@ -92,7 +92,9 @@ namespace Microsoft.Graph.ODataTemplateWriter.Extensions
             var referenceEntityTypes = new List<OdcmClass>();
             foreach (var referencePropertyType in referencePropertyTypes)
             {
-                var entityType = entityTypes.FirstOrDefault(entity => entity.Name == referencePropertyType.Name);
+                // use the FullName to match the types as there could be multiple types of the same name but in different namespaces
+                // e.g microsoft.graph.group and microsoft.graph.termstore.group
+                var entityType = entityTypes.FirstOrDefault(entity => entity.FullName == referencePropertyType.FullName);
 
                 if (entityType != null)
                 {

